ros/src/serow_ros/src/serow_driver.cpp

```diff
@@ -0,0 +1,166 @@
+
+                // Create the base imu measurement
+                const auto& imu_data = base_imu_data_.front();
+                serow::ImuMeasurement imu_measurement{};
+                imu_measurement.timestamp =
+                    static_cast<double>(imu_data.header.stamp.sec) +
+                    static_cast<double>(imu_data.header.stamp.nsec) * 1e-9;
+                imu_measurement.linear_acceleration =
+                    Eigen::Vector3d(imu_data.linear_acceleration.x, imu_data.linear_acceleration.y,
+                                    imu_data.linear_acceleration.z);
+                imu_measurement.angular_velocity =
+                    Eigen::Vector3d(imu_data.angular_velocity.x, imu_data.angular_velocity.y,
+                                    imu_data.angular_velocity.z);
+                base_imu_data_.pop();
+
+                // Create the leg F/T measurement
+                std::map<std::string, serow::ForceTorqueMeasurement> ft_measurements;
+                for (auto& [key, value] : ft_data_) {
+                    if (value.size()) {
+                        serow::ForceTorqueMeasurement ft{};
+                        const auto& ft_data = value.front();
+                        ft.timestamp = static_cast<double>(ft_data.header.stamp.sec) +
+                                    static_cast<double>(ft_data.header.stamp.nsec) * 1e-9;
+                        ft.force = Eigen::Vector3d(ft_data.wrench.force.x, ft_data.wrench.force.y,
+                                                ft_data.wrench.force.z);
+                        ft.torque = Eigen::Vector3d(ft_data.wrench.torque.x,
+                                                    ft_data.wrench.torque.y,
+                                                    ft_data.wrench.torque.z);
+                        ft_measurements[key] = std::move(ft);
+                        value.pop();
+                    }
+                }
+
+                serow_.filter(imu_measurement, joint_measurements,
+                            ft_measurements.size() > 0 ? std::make_optional(ft_measurements)
+                                                       : std::nullopt);
+
+                auto state_ = serow_.getState();
+                pose_estimate_.header.seq  += 1;
+                pose_estimate_.pose.position.x = state_->getBasePosition().x();
+                pose_estimate_.pose.position.y = state_->getBasePosition().y();
+                pose_estimate_.pose.position.z = state_->getBasePosition().z();
+                pose_estimate_.pose.orientation.x = state_->getBaseOrientation().coeffs().x();
+                pose_estimate_.pose.orientation.y = state_->getBaseOrientation().coeffs().y();
+                pose_estimate_.pose.orientation.z = state_->getBaseOrientation().coeffs().z();
+                pose_estimate_.pose.orientation.w = state_->getBaseOrientation().coeffs().w();
+                base_state_publisher_.publish(pose_estimate_);
```

Comment:
probably we want to publish base velocity and com position + velocity as well

Reply:
Will add it after we make sure that the base estimates are fixed